sdk/core/Microsoft.Azure.Core.Spatial/tests/Serialization/GeographyPointConverterTests.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Text.Json;
+using Azure.Core.Serialization;
+using Microsoft.Spatial;
+using NUnit.Framework;
+
+namespace Microsoft.Azure.Core.Spatial.Tests.Serialization
+{
+    public class GeographyPointConverterTests
+    {
+        [Test]
+        public void CanConvert()
+        {
+            GeographyPointConverter converter = new GeographyPointConverter();
+            Assert.IsTrue(converter.CanConvert(typeof(GeographyPoint)));
+        }
+
+        [Test]
+        public void Read()
+        {
+            JsonSerializerOptions options = new JsonSerializerOptions
+            {
+                Converters =
+                {
+                    new GeographyPointConverter(),
+                },
+            };
+
+            GeographyPoint point = JsonSerializer.Deserialize<GeographyPoint>(@"{""type"":""Point"",""coordinates"":[-121.726906,46.879967]}", options);
+
+            Assert.AreEqual(46.879967, point.Latitude);
+            Assert.AreEqual(-121.726906, point.Longitude);
+        }
+
+        [Test]
+        public void ReadMore()
+        {
+            JsonSerializerOptions options = new JsonSerializerOptions
+            {
+                Converters =
+                {
+                    new GeographyPointConverter(),
+                },
+            };
+
+            GeographyPoint point = JsonSerializer.Deserialize<GeographyPoint>(@"{""type"":""Point"",""coordinates"":[-121.726906,46.879967,2541.118],""crs"":{""type"":""name"",""properties"":{""name"":""EPSG:4326""}}}", options);
+
+            Assert.AreEqual(46.879967, point.Latitude);
+            Assert.AreEqual(-121.726906, point.Longitude);
+
+            // Not currently supported.
+            Assert.IsNull(point.Z);
+        }
+
+        [Test]
+        public void ReadIntegers()
+        {
+            JsonSerializerOptions options = new JsonSerializerOptions
+            {
+                Converters =
+                {
+                    new GeographyPointConverter(),
+                },
+            };
+
+            GeographyPoint point = JsonSerializer.Deserialize<GeographyPoint>(@"{""type"":""Point"",""coordinates"":[-121,46]}", options);
+
+            Assert.AreEqual(46.0, point.Latitude);
+            Assert.AreEqual(-121.0, point.Longitude);
+        }
+
+        [Test]
+        public void ReadCaseInsensitive()
+        {
+            JsonSerializerOptions options = new JsonSerializerOptions
+            {
+                Converters =
+                {
+                    new GeographyPointConverter(),
+                },
+                PropertyNameCaseInsensitive = true,
+            };
+
+            GeographyPoint point = JsonSerializer.Deserialize<GeographyPoint>(@"{""Type"":""point"",""Coordinates"":[-121.726906,46.879967]}", options);
```

Comment:
This is against the spec. Have you seen a scenario that requires this kind of relaxed parsing?

Reply:
I realize that and originally considered not doing this, schemas don't always follow the specs. I figured the support for this is easy enough *and* you have to opt into it (default is case-sensitive), so it's probably worth at least supporting. Or would you prefer I remove it and do only case-sensitive?

Reply:
I would follow the spec unless there is a reason not to.